invenio_rdm_records/resources/serializers/marcxml/__init__.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""MARCXML Serializer for Invenio RDM Records."""
+
+
+import traceback
+from copy import deepcopy
+
+from dojson.contrib.to_marc21.utils import dumps_etree
+from flask_resources.serializers import SerializerMixin
+from lxml import etree
+
+from .schema import MARCXMLSchema
+
+
+class MARCXMLSerializer(SerializerMixin):
```

Comment:
Would it make sense to inherit from `MarshmallowSerializer`? or does it complicate a lot the rest of the logic?
https://github.com/inveniosoftware/flask-resources/blob/master/flask_resources/serializers/base.py#L31

Reply:
The choice to inherit from `SerializerMixin` was made because I noticed a pattern on the other serializers
if return type is `JSON` --> inherit from `MarshmallowJSONSerializer` (which in turn inherits `MarshmallowSerializer`)
if return type is `XML` --> inherit from `SerializerMixin`
Nothing seems to directly inherit `MarshmallowSerializer`
Note : `MarshmallowSerializer` inherits `SerializerMixin`

Reply:
I see your point, for example with the Dublin Core serializer. However, I think we made a mistake there. Analyzing a bit the inheritance in the serializers:

- We have to base classes: a raw serializer `SerializerMixin` which is nothing more than an interface (could just be an abstract class instead of a Mixing since it does not bring functionality, but that is another story). Then we have [MarshmallowSerializer](https://github.com/inveniosoftware/flask-resources/blob/master/flask_resources/serializers/base.py#L31), which is meant for those serializers where we use a Marshmallow schema (which is the case here and also in DublinCore).
- Then we have specific ones for JSON, i.e. `MarshmallowJSONSerializer`, which basically takes the `MarshmallowSerializer` but [specifies an output format](https://github.com/inveniosoftware/flask-resources/blob/master/flask_resources/serializers/json.py#L85): JSON.

In my head the ideal case scenario would be:
1. We create a `MarshmallowXMLSerializer` that takes the schema and then outputs XML.
2. We migrate DublinCore to use this

However, your code is already functional and I do not want to block you, meaning we could potentially merge and do more changes in another PR. Point number 2 is definetly outside the scope of this PR, I'm wondering about point 1... WDYT? cc: @ntarocco

Reply:
Opened an issue about this https://github.com/inveniosoftware/invenio-rdm-records/issues/1184